crates/librqbit_core/src/magnet.rs

```diff
@@ -17,15 +55,20 @@ impl Magnet {
         if url.scheme() != "magnet" {
             anyhow::bail!("expected scheme magnet");
         }
-        let mut info_hash: Option<Id20> = None;
+        let mut info_hash: Option<InfoHash> = None;
         let mut trackers = Vec::<String>::new();
         for (key, value) in url.query_pairs() {
             match key.as_ref() {
-                "xt" => match value.as_ref().strip_prefix("urn:btih:") {
-                    Some(infohash) => {
-                        info_hash.replace(Id20::from_str(infohash)?);
+                "xt" => {
+                    if let Some(ih) = value.as_ref().strip_prefix("urn:btih:") {
+                        let i: InfoHash = InfoHash::V1(Id20::from_str(ih)?);
+                        info_hash.replace(i);
```

Comment:
this "replace" doesn't account for both hashes being possible in the URL. I guess there's nothing that prevents both v1 and v2 hashes being possible in the URL. I don't know if there's a spec about it, but intuitively it makes total sense to have both in the same URL.